tests/test_stnode.py

```diff
@@ -104,14 +107,30 @@ def test_wfi_mode():
 
 
 @pytest.mark.parametrize("node_class", stnode.NODE_CLASSES)
-def test_serialization(node_class, tmp_path):
+def test_serialization(node_class, tmp_path, request):
     file_path = tmp_path / "test.asdf"
 
     node = node_class.create_fake_data()
     with asdf.AsdfFile() as af:
         af["node"] = node
         af.write_to(file_path)
 
+    if node_class is stnode.MosaicBasic:
+        # See issue https://github.com/spacetelescope/rad/issues/694
+        # Essentially the product_type is being accidentally set to the `ProductType` tagged
+        #   scalar node, due to a name collision with the `product_typ` tagged schema. This
+        #   means that weird things are happening because when read the `MosaicBasic` node
+        #   will have `read_tag` populated and since it is a pre-internal tag removal schema
+        #   it will run the wrapping code that causes the `product_type` to be autop wrapped
+        #   into the `ProductType` tagged scalar node which means it is not longer a simple string
+        #   like it really should be. `create_fake_data` is not inducing the wrapper because
+        #   it is not setting the `read_tag` attribute, so auto wrapping is not happening.
+        # This is technically a subtle RDM bug for the old internal tagged objects, but this is the
+        #   only case and it would only effect things taking advantage of the old code.
+        request.applymarker(
+            pytest.mark.xfail(reason="mosaic basic has a product_type not derived from the same schema as product_type")
+        )
```

Comment:
It should be possible to resolve this bug, but it is effectively only detectable at this level since the `ProductType` object is simply a `str` subclass with some metadata attributes added.

After trying to solve this cleanly, I don't think it is worth the effort or complexity necessary to correct this. A better solution would be to do as the linked RAD issue suggests and align the two `product_type` definitions so that it can be a single direct reference.

Reply:
Would you remove this xfail and update the following line:
https://github.com/spacetelescope/roman_datamodels/blob/69d0823f2875f6ed1a855e35cbf935c5ad0e0d7b/src/roman_datamodels/stnode/_stnode.py#L66
to
```python
        if 'mosaic_basic' not in base and manifest_uri == _LAST_INTERNAL_TAGGED_MANIFEST:
```
It looks like the current code on main is incorrectly wrapping product_type for old `MosaicBasic` instances. From the schemas this should never be wrapped so there is no need to add a wrap limit for that schema/node.

Reply:
Done